monai/metrics/seg_metric_utils.py

```diff
@@ -0,0 +1,220 @@
+            n_classes = input_data.shape[1]
+            assert n_classes == len(bin_threshold), "{}".format(error_hint)
+            for cls_num in range(n_classes):
+                input_data[:, cls_num] = (input_data[:, cls_num] >= bin_threshold[cls_num]).float()
+        else:
+            input_data = (input_data >= bin_threshold).float()
+    elif bin_mode == "mutually_exclusive":
+        assert input_ndim > 1, "mutually_exclusive is used for multi-class tasks."
+        n_classes = input_data.shape[1]
+        assert n_classes > 1, "mutually_exclusive is used for multi-class tasks."
+        input_data = torch.argmax(input_data, dim=1, keepdim=True)
+        input_data = one_hot(input_data, num_classes=n_classes)
+    return input_data
+
+
+def preprocess_input(
+    y_pred: torch.Tensor,
+    y: torch.Tensor,
+    to_onehot_y: bool = False,
+    activation: Optional[Union[str, Callable]] = None,
+    bin_mode: Optional[str] = None,
+    bin_threshold: Union[float, Sequence[float]] = 0.5,
+    include_background: bool = True,
+):
+    """
+    Args:
+        y_pred: predictions. As for classification tasks,
+            `y_pred` should has the shape [BN] where N is larger than 1. As for segmentation tasks,
+            the shape should be [BNHW] or [BNHWD].
+        y: ground truth, the first dim is batch.
+        to_onehot_y: whether to convert `y` into the one-hot format. If not to convert the format, ``y``
+        should has the same shape as ``y_pred``. Defaults to False.
+        activation: [``None``, ``"sigmoid"``, ``"softmax"``]
+            Activation method, if specified, an activation function will be employed for `y_pred` before
+            computation. Defaults to None. The parameter can also be a callable function, for example:
+            ``activation = lambda x: torch.log_softmax(x)``.
+        bin_mode: [``None``, ``"threshold"``, ``"mutually_exclusive"``]
+            Binarization method, if specified, a binarization manipulation will be employed for `y_pred` before
+            computation. Defaults to None.
+
+            - ``"threshold"``, a single threshold or a sequence of thresholds should be set.
+            - ``"mutually_exclusive"``, `y_pred` will be converted by a combination of `argmax` and `to_onehot`.
+        bin_threshold: the threshold for binarization, can be a single value or a sequence of
+            values that each one of the value represents a threshold for a class.
+        include_background: whether to include computation on the first channel of the predicted output.
+            Defaults to True.
+
+    Raises:
+        AssertionError: When ``y_pred`` only has one dimebsion.
+        AssertionError: When ``y`` and ``y_pred`` have different shapes before output.
+
+    """
+
+    # convert y into one-hot version if needed
+    assert y_pred.ndimension() > 1, "y_pred should at least has two dimensions batch and n_classes."
```

Comment:
In a number of places asserts are being used to check inputs. I recommend that input validation be done with conditional check followed by a raising of a `ValueError`. This is inline with standard practice and is what users expect in terms of the type of exception they would get, plus there's the issue that assertions can be turned off at runtime and input validation checks shouldn't ever be disabled.